pages/lessons/projects/6.mdx

```diff
@@ -0,0 +1,796 @@
+
+The Ethereum blockchain (and most EVM-compatible blockchains) has a native token
+that is used to pay for transaction costs, and also to reward validators in
+Proof of Stake, or previously miners in Proof of Work. In Ethereum that native
+token is Ether (or ETH), in Polygon its MATIC, and so on.
+
+Let's focus on the Ethereum blockchain for the sake of clarity.
+
+Every _other_ Token or "Crypto" you see, use or interact with in Ethereum is
+basically a smart contract that lets you send this tokens, receive them and
+check your balance.
+
+Stablecoins like **DAI**, Tether (**USDT**), USD Coin (**USDC**) or tokens from
+projects like Uniswap (**UNI**), MakerDAO (**MKR**), Basic Attention Token
+(**BAT**) are all smart contracts that follow a standard, the ERC20 standard.
+
+## Why do we want to create yet another Token?
+
+A lot of reasons, but here are some:
+
+- We are learning and we want to know how the ecosystem works
+- We have a project or DAO that needs some means of governance
+- We have a community that wants to be a part, take some ownership or support
+  our project
+- We need a token with extra functionality that doesn't exist yet
+
+The possibilities are endless.
+
+## What do we need to do with our Tokens?
+
+Some basic functionality that our tokens need to provide are:
+
+- Transfer tokens from one account to another
+- Get the current token balance for an account
+- Get the total supply of tokens available in the network
+- Approve an amount of tokens from an account to be spent by a third-party
+  account
+
+<SideDrawer buttonText="More info on the ERC20 Standard">
+
+To achive the functionality needed for an ERC20 token, the standard requires us
+to implement 9 methods and 2 events.
+
+Here is a quick overview of what our contracts needs to be called an ERC20 Token
+contract:
+
+```solidity
+// Methods:
+function name() public view returns (string)
+function symbol() public view returns (string)
+function decimals() public view returns (uint8)
+function totalSupply() public view returns (uint256)
+function balanceOf(address _owner) public view returns (uint256 balance)
+function transfer(address _to, uint256 _value) public returns (bool success)
+function transferFrom(address _from, address _to, uint256 _value) public returns (bool success)
+function approve(address _spender, uint256 _value) public returns (bool success)
+function allowance(address _owner, address _spender) public view returns (uint256 remaining)
+
+// Events:
+event Transfer(address indexed _from, address indexed _to, uint256 _value)
+event Approval(address indexed _owner, address indexed _spender, uint256 _value)
+```
+
+You can check the details in the full EIP-20 titled
+[ERC-20: Token Standard](https://eips.ethereum.org/EIPS/eip-20#token)
+
+</SideDrawer>
+
+## Has someone created a battle-tested library for the standard?
+
+Yes. OpenZeppelin (among others) has created an implementation we can inherit to
+create
+our own tokens easily.
+
```

Comment:
Might a deeper conversation be made here about what "inherit" means in this context. What functionality is extracted from the Open Zeppelin contracts? @7i7o I know we might have covered this in Fundamentals - Open Zeppelin, but I just feel that learners may have a lingering question of "What benefit is coming from the Open Zeppelin contracts? Why is this so important?".